src/install.js

```diff
@@ -46,15 +46,46 @@ questions.optional = [
 	},
 ];
 
-function checkSetupFlag() {
-	let setupVal = install.values;
+function checkSetupFlagEnv() {
+	let setupVal = install.values || {};
+
+	const envConfMap = {
+		NODEBB_URL: 'url',
+		NODEBB_PORT: 'port',
+		NODEBB_ADMIN_USERNAME: 'admin:username',
+		NODEBB_ADMIN_PASSWORD: 'admin:password',
+		NODEBB_ADMIN_EMAIL: 'admin:email',
+		NODEBB_DB: 'database',
+		NODEBB_DB_HOST: 'host',
+		NODEBB_DB_PORT: 'port',
+		NODEBB_DB_USER: 'username',
+		NODEBB_DB_PASSWORD: 'password',
+		NODEBB_DB_NAME: 'database',
+		NODEBB_DB_SSL: 'ssl',
+	};
+
+	// Set setup values from env vars (if set)
+	winston.info('checking env vars for setup info...');
+
+	Object.entries(process.env).forEach(([evName, evValue]) => { // get setup values from env
+		if (evName.startsWith('NODEBB_DB_')) {
+			setupVal[`${process.env.NODEBB_DB}:${envConfMap[evName]}`] = evValue;
+		} else if (evName.startsWith('NODEBB_')) {
+			setupVal[envConfMap[evName]] = evValue;
+		}
+	});
+
+	setupVal['admin:password:confirm'] = setupVal['admin:password'];
 
+	// try to get setup values from json, if successful this overwrites all values set by env
+	// TODO: better behaviour would be to support overrides per value, i.e. in order of priority (generic pattern):
+	//       flag, env, config file, default
 	try {
 		if (nconf.get('setup')) {
 			setupVal = JSON.parse(nconf.get('setup'));
```

Comment:
```
const setupJSON = JSON.parse(nconf.get('setup'));
setupVal = { ...setupVal, ...setupJSON };
```

Perhaps? Untested.

Reply:
done.
Indeed this works, I tested it with combinations of incomplete/wrong `json` and incomplete/wrong env vars.
If both env and `json` have values, the `json` prevails.